controllers/greptimedbcluster/deployers/datanode.go

```diff
@@ -436,11 +452,36 @@ func (d *DatanodeDeployer) generatePodTemplateSpec(cluster *v1alpha1.GreptimeDBC
 				},
 			},
 		},
-	})
+	}
 
-	podTemplateSpec.ObjectMeta.Labels = deployer.MergeStringMap(podTemplateSpec.ObjectMeta.Labels, map[string]string{
-		GreptimeComponentName: d.ResourceName(cluster.Name, v1alpha1.DatanodeComponentKind),
-	})
+	if cluster.Spec.StorageProvider != nil {
+		if cluster.Spec.StorageProvider.Local != nil {
+			storageArgs := []string{
+				"--storage-type", "Local",
+				"--local-storage-dir", cluster.Spec.StorageProvider.Local.Directory,
+			}
+			initializer.Args = append(initializer.Args, storageArgs...)
+		}
 
-	return podTemplateSpec
+		if cluster.Spec.StorageProvider.S3 != nil {
```

Comment:
if localstorage != nil && s3storage != nil, what to do.

Reply:
@daviderli614 I already add the validation in `validate()`, if the case happen, controller will reject the creation.